xrpl/utils/hash_utils.py

```
@@ -0,0 +1,287 @@
+    addr2_hex = address_to_hex(address2)
+    
+    # Ensure consistent ordering (lower address first)
+    if addr1_hex > addr2_hex:
+        addr1_hex, addr2_hex = addr2_hex, addr1_hex
+    
+    # Handle currency formatting (must be 20 bytes)
+    if isinstance(currency, str) and len(currency) == 3:
+        currency_hex = ("00" * 12) + currency.encode("ascii").hex().upper() + ("00" * 5)
+    elif isinstance(currency, bytes):
+        if len(currency) != 20:
+            raise ValueError("Currency bytes must be exactly 20 bytes.")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)